### PR TITLE
[MAINT] change function privacy in _utils

### LINKED ISSUE #3628: make sure public / private functions in _utils to start without / with a leading underscore
From what I understood anything in the _utils folder is considered "private" so the leading underscore does not have to be systematic

Yes I'd say that that is the case

> I am completely fine trying to homogenize this to all leading-underscore which would be nice because private / non-private status of the function would be in the filename rather than in the parent folder where its module can be found.

Indeed let's follow up on that in a separate issue as it's not really related to this PR

_Originally posted by @ymzayek in https://github.com/nilearn/nilearn/pull/3525#discussion_r1151533681_

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Relates to #2750 and #3319
- Closes #3628

Changes proposed in this pull request:

- ensure that private functions to the module have leading underscore
- ensure that public functions to the module have no leading underscore

### CODE REVIEW

File: nilearn/_utils/tests/test_niimg_conversions.py [hunk 1]
```diff
@@ -369,19 +369,19 @@ def test_check_niimg_wildcards_no_expand_wildcards(
     ni.EXPAND_PATH_WILDCARDS = True
 
 
-def test_iter_check_niimgs_error():
+def testiter_check_niimgs_error():
```
Comment (Remi-Gau) [inline nilearn/_utils/tests/test_niimg_conversions.py:372]:
```suggestion
def test_iter_check_niimgs_error():
```

File: nilearn/_utils/tests/test_param_validation.py [hunk 2]
```diff
@@ -64,10 +64,10 @@ def test_check_threshold():
     )
 
 
-def test_get_mask_volume():
+def testget_mask_volume():
```
Comment (Remi-Gau) [inline nilearn/_utils/tests/test_param_validation.py:67]:
```suggestion
def test_get_mask_volume():
```

File: nilearn/plotting/displays/_slicers.py [hunk 3]
```diff
@@ -9,7 +9,7 @@
 
 from nilearn._utils import check_niimg_3d
 from nilearn._utils.docs import fill_doc
-from nilearn._utils.niimg import _is_binary_niimg, _safe_get_data
+from nilearn._utils.niimg import _safe_get_data, is_binary_niimg
```
Comment (Remi-Gau) [inline nilearn/plotting/displays/_slicers.py:12]: rename `_safe_get_data`

**Review by bthirion — COMMENTED** [review]
LGTM overall.

File: nilearn/_utils/tests/test_niimg_conversions.py [hunk 4]
```diff
@@ -374,14 +374,14 @@ def test_iter_check_niimgs_error():
...
+            list(iter_check_niimg(empty))
 
     nofile_path = "/tmp/nofile"
     with pytest.raises(ValueError, match=no_file_matching % nofile_path):
-        list(_iter_check_niimg(nofile_path))
+        list(iter_check_niimg(nofile_path))
 
 
-def test_iter_check_niimgs(tmp_path, img_4d_zeros_eye):
+def testiter_check_niimgs(tmp_path, img_4d_zeros_eye):
```
[the remark this answers is not in the document]
Comment (Remi-Gau) [reply nilearn/_utils/tests/test_niimg_conversions.py:384]: tgat was a typo
thanks for the catch

Comment (Remi-Gau) [inline nilearn/_utils/tests/test_niimg_conversions.py:384]:
```suggestion
def test_iter_check_niimgs(tmp_path, img_4d_zeros_eye):
```

**Comment by Remi-Gau** [comment]
apparently we get some errors in doc build following this one.

**Pull request merged**